### ISSUE TITLE
oxlint config file not work on Windows

### ISSUE THREAD

**Issue by EYHN** [oxc-project/oxc#8188]
When I try to use the same config file on both Windows and Linux, the config file on Windows does not work.

Reproduction repo:

https://github.com/EYHN/oxlinttest

command:

```bash
npx oxlint -c oxlint.json
```

Windows:

![image](https://github.com/user-attachments/assets/97fab631-11c7-47c4-afa7-8ac67487dc24)

Linux:

![Screenshot 2024-12-30 112527](https://github.com/user-attachments/assets/7380e7d2-b6de-4fb7-afe9-56c8bd646a8d)

**Comment by camc314** [comment]
This looks like it's behaving as intended?

you've got `src/bad.js` in your ignorepatterns hence the file is not being listed

https://github.com/EYHN/oxlinttest/blob/598eb43514a26dfb0c674f5f188a4c7c6fe068a3/oxlint.json#L2-L4

Using your repro, if i remove that ignore pattern, it works as expected

**Comment by Sysix** [comment]
> you've got src/bad.js in your ignorepatterns hence the file is not being listed

This is the problem under windows. Linux ignores it, windows will lint it.

**Comment by camc314** [comment]
ah i see - thanks

i guess we're joining paths in a platform sensitive way somewhere

**Comment by camc314** [comment]
Will need someone with a windows machine to debug this.

One of the things that appears to be happening is that the cli argument does not seem to be picked up correctly

**Comment by Sysix** [comment]
I can look into this, I have window with wsl2 (window subsystem (linux)).
But in this year :')
